app/models/course.rb

```diff
@@ -42,6 +42,11 @@ class Course < ApplicationRecord
     open: 1
   }, _prefix: :application_status
 
+  enum course_type: {
+    postgraduate: 0,
+    undergraduate: 1
+  }, _prefix: :type
+
```

Comment:
How do you feel about the methods being named `course.type_postgraduate?` and `course.type_undergraduate?`course.alternatively we could have `course.postgraduate_course_type?`, with a `course_type` suffix, or even `course.postgraduate?` without a prefix or suffix.

Reply:
I'd prefer by far `course.postgraduate?` simple, direct, clear and there won't be any clashes.

Reply:
`course.postgraduate?` would definitely be my preference too but unfortunately it causes a naming clash:

`ArgumentError: You tried to define an enum named "course_type" on the model "Course", but this will generate a instance method "postgraduate?", which is already defined by another enum. (ArgumentError)`

I've gone with your suggestion @dcyoung-dev `course.postgraduate_course_type?` I think that sounds a little nicer than what I had.

Reply:
That is a very interesting error - what other enum is generating a `postgraduate?` method 🧐

Reply:
UCAS strikes again - https://github.com/DFE-Digital/publish-teacher-training/blob/bcda42d57e6ef04931fef14a493dfef3e7179375/app/models/concerns/with_qualifications.rb#L30

I'm happy to introduce `course.postgraduate_course_type?` with the hope we can remove this other enum at some point